### PR TITLE
Updated with issue fixes, code cleanup and other improvements

### PR DESCRIPTION
**Pull request by praveenraj2018**
**Improvements includes:**

- Local modules have been replaced with nf-core modules, except haplotypeCaller as it needs to run on split interval files
- Updated all tools to latest one from repo (for star, gatk4, picard, cat, samtools, untar, etc)
- Issue with cat_fastq has been fixed
- Code clean up - removed unwanted modules and code blocks
- Fixing of all lint errors and most of the warinings
- Tested on GM12878 sample data set as well as standard RNAseq dataset
- Tested on sample with multiple lanes/replicate to check cat_fastq
- Oragnized output directory structure with files
- MultiQC report
- Versions.yml for all local modules / subworkflows
- Schema json has been updated with all parameters to avoid runtime warnings as seen earlier

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: bin/check_samplesheet.py [hunk 1]
```diff
@@ -7,14 +7,15 @@
 
 
 def parse_args(args=None):
-    Description = "Reformat nf-core/rnavar samplesheet file and check its contents."
+    Description = "Reformat nf-core/rnaseq samplesheet file and check its contents."
```
Comment (maxulysse) [inline bin/check_samplesheet.py:10]:
```suggestion
    Description = "Reformat nf-core/rnavar samplesheet file and check its contents."
```

**Review by maxulysse — APPROVED** [review]
I love it

**Comment by praveenraj2018** [comment]
@maxulysse Thanks for the review. 
I was looking at the linting failure and it looks like the version of nf-core is older (nf-core/tools version 2.1) than I have (2.2.dev0) - I think 2.2.dev0 is recommended. I don't see any errors at my side with 2.2.dev0. Can you help to fix it?

**Comment by maxulysse** [comment]
I think it's fine is the linting is failing for now, we're still doing some improvements on the modules syntax and then we'll have a proper `tools` release.
So either we wait for this `tools` release and then sync everything and fix the lint, and then we can finally start working on the release here.
Or we can think of something else.
In any case, I think we're good given the current linting errors

**Comment by praveenraj2018** [comment]
Great! Thanks @maxulysse

**Pull request merged**